lib/livebook/application.ex

```diff
@@ -197,12 +197,14 @@ defmodule Livebook.Application do
     port = Livebook.Config.iframe_port()
 
     if server? do
-      iframe_opts = [
-        scheme: :http,
-        plug: LivebookWeb.IframeEndpoint,
-        options: [port: port],
-        ip: Application.fetch_env!(:livebook, LivebookWeb.Endpoint)[:http][:ip]
-      ]
+      livebook_ip = Application.fetch_env!(:livebook, LivebookWeb.Endpoint)[:http][:ip]
+
+      iframe_opts =
+        [
+          scheme: :http,
+          plug: LivebookWeb.IframeEndpoint,
+          options: [port: port]
+        ] ++ if(livebook_ip, do: [ip: livebook_ip], else: [])
 
```

Comment:
```suggestion
      http = Application.fetch_env!(:livebook, LivebookWeb.Endpoint)[:http]

      iframe_opts =
        [
          scheme: :http,
          plug: LivebookWeb.IframeEndpoint,
          options: [port: port]
        ] ++ Keyword.take(http, [:ip])

```

Reply:
@josevalim actually, shouldn't both `:port` and `:ip` be top-level options? From [here](https://github.com/elixir-plug/plug_cowboy/blob/v2.5.2/lib/plug/cowboy.ex#L232-L237):

```elixir
    # We support :options for backwards compatibility.
    cowboy_opts =
      opts
      |> Keyword.drop([:scheme, :plug, :options])
      |> Kernel.++(Keyword.get(opts, :options, []))
```